lib/asset_writer.rb

```diff
@@ -0,0 +1,37 @@
+class AssetWriter
+  attr_reader :asset, :target_path
+
+  delegate :logical_path, :digest_path, :to => :asset
+
+  def initialize(asset_or_path, target_dir)
+    @asset = find_asset(asset_or_path)
```

Comment:
Overall this approach looks good, but I'm curious here about the coupling between finding an asset and calling the digest_path 2 lines down.  Also why is this an asset_or_path?  Is that just for test purposes?